File: fix_partner_profile.py
import re

with open('apps/web/app/[locale]/fixers/page.tsx', 'r') as f:
    content = f.read()

# Locate the end of PartnerProfile function or just before the final return closing tag
# Let's insert the AI assessment section before the Settings / Logout buttons in PartnerProfile

to_insert = """
      {/* AI Assessment & Profile Completion */}
      <div className="bg-white rounded-2xl shadow-sm border border-gray-100 p-6">
        <h3 className="text-lg font-bold text-gray-900 mb-4 flex items-center gap-2">🤖 {locale === "th" ? "การประเมินระดับ CBLUE AI" : locale === "zh" ? "CBLUE AI 等级评估" : "CBLUE AI Tier Assessment"}</h3>
        <div className="flex flex-col md:flex-row gap-6">
          <div className="flex-1 bg-gradient-to-br from-indigo-50 to-purple-50 rounded-xl p-5 border border-indigo-100">
            <div className="flex justify-between items-start mb-4">
              <div>
                <span className="text-sm font-bold text-indigo-600 uppercase tracking-wider">{locale === "th" ? "ระดับปัจจุบัน" : locale === "zh" ? "当前等级" : "Current Tier"}</span>
                <div className="text-3xl font-extrabold text-gray-900 mt-1">Specialist</div>
              </div>
              <div className="text-right">
                <span className="text-sm text-gray-500">{locale === "th" ? "คะแนนรวม" : locale === "zh" ? "总分" : "Overall Score"}</span>
                <div className="text-2xl font-bold text-indigo-700">69<span className="text-sm text-gray-400">/100</span></div>
              </div>
            </div>
            
            <div className="w-full bg-indigo-200/50 rounded-full h-2.5 mb-2">
              <div className="bg-indigo-600 h-2.5 rounded-full" style={{ width: "69%" }}></div>
            </div>
            <p className="text-xs text-indigo-800 flex items-center gap-1.5 font-medium mt-3">
              <span>⚠️</span> {locale === "th" ? "ยืนยันบางส่วน — เติมโปรไฟล์ให้สมบูรณ์เพื่อเลื่อนระดับ" : locale === "zh" ? "部分验证 — 完善个人资料以提升等级" : "Partially Verified — Complete profile to improve"}
            </p>
          </div>

          <div className="flex-[2] grid grid-cols-1 sm:grid-cols-2 gap-x-6 gap-y-3 text-sm">
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "ประสบการณ์" : locale === "zh" ? "经验" : "Experience"}</span><span className="font-medium text-green-600">25/25</span></div>
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "ความหลากหลายของทักษะ" : locale === "zh" ? "技能广度" : "Skills Breadth"}</span><span className="font-medium text-green-600">12/15</span></div>
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "การยืนยันตัวตน KYC" : locale === "zh" ? "KYC身份验证" : "KYC Verification"}</span><span className="font-medium text-green-600">15/15</span></div>
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "ผลงานและหลักฐาน" : locale === "zh" ? "作品集与证据" : "Portfolio & Evidence"}</span><span className="font-medium text-red-500">0/15</span></div>
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "ความสมบูรณ์ของโปรไฟล์" : locale === "zh" ? "个人资料完整度" : "Profile Completeness"}</span><span className="font-medium text-amber-500">7/10</span></div>
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "ตารางราคา" : locale === "zh" ? "价格表" : "Price List"}</span><span className="font-medium text-amber-500">6/10</span></div>
            <div className="flex justify-between items-center"><span className="text-gray-600">{locale === "th" ? "การตรวจสอบใบรับรอง" : locale === "zh" ? "凭证验证" : "Credential Verification"}</span><span className="font-medium text-amber-500">4/10</span></div>
          </div>
        </div>

        <div className="mt-6 pt-5 border-t border-gray-100">
          <h4 className="text-sm font-bold text-gray-900 mb-3 flex items-center gap-2">🔍 {locale === "th" ? "ผลการตรวจสอบ AI" : locale === "zh" ? "AI验证结果" : "AI Verification Results"}</h4>
          <ul className="text-sm space-y-2 mb-5">
            <li className="flex items-start gap-2 text-gray-600"><span className="text-red-500 mt-0.5">❌</span> {locale === "th" ? "ไม่ได้ระบุข้อมูลบริษัท" : locale === "zh" ? "未提供公司信息" : "No company info provided"}</li>
            <li className="flex items-start gap-2 text-gray-600"><span className="text-green-500 mt-0.5">✅</span> {locale === "th" ? "ประสบการณ์สอดคล้องกับประเภทงาน" : locale === "zh" ? "经验与项目类型一致" : "Experience consistent with project type"}</li>
            <li className="flex items-start gap-2 text-gray-600"><span className="text-red-500 mt-0.5">❌</span> {locale === "th" ? "ไม่ได้ระบุรายละเอียดงาน" : locale === "zh" ? "未提供工作描述" : "No work description provided"}</li>
            <li className="flex items-start gap-2 text-gray-600"><span className="text-green-500 mt-0.5">✅</span> {locale === "th" ? "เอกสาร KYC ครบถ้วน (หน้า-หลัง)" : locale === "zh" ? "KYC文件齐全（正反面）" : "KYC documents complete (front & back)"}</li>
          </ul>

          <div className="bg-blue-50 rounded-xl p-4 mb-4">
            <h4 className="text-sm font-bold text-blue-900 flex items-center gap-2 mb-1">💡 {locale === "th" ? "วิธีเลื่อนระดับ" : locale === "zh" ? "如何升级" : "How to upgrade"}</h4>
            <p className="text-xs text-blue-800 leading-relaxed">
              {locale === "th" ? "รับประสบการณ์เพิ่ม อัปโหลดผลงาน อัปเดตใบรับรอง และรักษาคะแนนรีวิวที่ดี — CBLUE AI จะประเมินและเลื่อนระดับคุณอัตโนมัติเมื่อคุณแก้ไขโปรไฟล์หรือสะสมประวัติการทำงาน" : locale === "zh" ? "积累更多经验、上传作品集、更新认证并保持良好的评价 —— 当您编辑个人资料或积累工作历史时，CBLUE AI 将自动重新评估并提升您的等级。" : "Gain more experience, upload portfolio work, update certifications, and maintain good reviews — CBLUE AI will automatically re-evaluate and upgrade your tier when you edit your profile or accumulate work history."}
            </p>
          </div>
          
          <div className="bg-gray-50 rounded-xl p-4 flex items-start gap-3">
            <span className="text-xl">🔒</span>
            <p className="text-xs text-gray-500 leading-relaxed">
              <strong className="text-gray-700">{locale === "th" ? "ความปลอดภัย:" : locale === "zh" ? "安全:" : "Security:"}</strong> {locale === "th" ? "ข้อมูลของคุณถูกเข้ารหัสและปกป้องภายใต้ PDPA มีการตรวจสอบใบรับรองเพื่อรักษามาตรฐานของแพลตฟอร์ม" : locale === "zh" ? "您的数据已加密并受PDPA保护。验证凭据以维护平台完整性。" : "Your data is encrypted and protected under PDPA. Credentials are verified to maintain platform integrity."}
            </p>
          </div>
        </div>

        <div className="mt-6 flex gap-3">
          <Link href={`${prefix}/fixers/register`} className="flex-1 py-3 bg-indigo-600 hover:bg-indigo-700 text-white text-center font-bold text-sm rounded-xl transition shadow">
            {locale === "th" ? "แก้ไขโปรไฟล์" : locale === "zh" ? "编辑个人资料" : "Edit Profile"}
          </Link>
        </div>
      </div>
"""

# Let's insert it before the last settings block
# search for `{/* Settings */}` in the file
content = content.replace('{/* Settings */}', to_insert + '\n      {/* Settings */}')

with open('apps/web/app/[locale]/fixers/page.tsx', 'w') as f:
    f.write(content)

File: fix_partner_profile.js
const fs = require('fs');

let code = fs.readFileSync('apps/web/app/[locale]/fixers/page.tsx', 'utf8');

const enhancedProfile = `
/* ===== PARTNER PROFILE ===== */
function PartnerProfile({ locale, prefix, partner }: { locale: string; prefix: string; partner: PartnerInfo | null }) {
  const router = useRouter();
  if (!partner) {
    return (
      <div className="bg-white rounded-2xl shadow-sm border border-gray-100 p-10 text-center">
        <div className="text-5xl mb-4">👤</div>
        <h2 className="text-xl font-bold text-gray-900 mb-2">{locale === "th" ? "เข้าสู่ระบบเพื่อดูโปรไฟล์" : locale === "zh" ? "登录查看个人资料" : "Log in to view profile"}</h2>
        <p className="text-sm text-gray-500 mb-6">{locale === "th" ? "เข้าสู่ระบบเพื่อจัดการข้อมูลและการตั้งค่า" : locale === "zh" ? "登录管理您的合作伙伴账户" : "Sign in to manage your partner account"}</p>
        <Link href={\`\${prefix}/subscription/login?redirect=/fixers\`} className="px-6 py-2.5 bg-sky-600 text-white rounded-lg font-bold hover:bg-sky-700 transition inline-block">
          {locale === "th" ? "เข้าสู่ระบบ" : locale === "zh" ? "登录" : "Log In"}
        </Link>
      </div>
    );
  }

  return (
    <div className="space-y-6">
      {/* Profile Overview Card */}
      <div className="bg-white rounded-2xl shadow-sm border border-gray-100 p-8">
        <div className="flex flex-col md:flex-row gap-8 items-start">
          <div className="w-32 h-32 rounded-2xl bg-gradient-to-br from-purple-100 to-indigo-50 flex items-center justify-center shadow-inner flex-shrink-0 relative">
            <span className="text-5xl">👤</span>
            <div className="absolute -bottom-2 -right-2 bg-white rounded-full p-1 shadow">
              <span className="w-8 h-8 flex items-center justify-center bg-green-100 text-green-600 rounded-full text-xs font-bold">✓</span>
            </div>
          </div>
          
          <div className="flex-1 w-full">
            <div className="flex justify-between items-start mb-4">
              <div>
                <h2 className="text-2xl font-bold text-gray-900">{partner.name}</h2>
                <div className="flex items-center gap-2 mt-1">
                  <span className="text-sm font-semibold text-purple-700 bg-purple-100 px-2 py-0.5 rounded">Specialist Tier</span>
                  <span className="text-xs text-gray-500 flex items-center gap-1"><span className="text-green-500">✓</span> {locale === "th" ? "ยืนยันตัวตนแล้ว (KYC)" : "Verified (KYC)"}</span>
                </div>
              </div>
              <button className="px-5 py-2.5 bg-gray-900 text-white rounded-lg hover:bg-gray-800 transition text-sm font-semibold shadow-sm">
                {locale === "th" ? "แก้ไขโปรไฟล์" : locale === "zh" ? "编辑资料" : "Edit Profile"}
              </button>
            </div>

            <div className="grid grid-cols-1 sm:grid-cols-3 gap-x-8 gap-y-4 pt-4 border-t border-gray-100">
              <div>
                <h3 className="text-xs font-semibold text-gray-400 uppercase tracking-wider mb-1">{locale === "th" ? "อีเมล" : "Email"}</h3>
                <p className="text-gray-900 font-medium text-sm truncate">{partner.email}</p>
              </div>
              <div>
                <h3 className="text-xs font-semibold text-gray-400 uppercase tracking-wider mb-1">{locale === "th" ? "เบอร์โทรศัพท์" : "Phone"}</h3>
                <p className="text-gray-900 font-medium text-sm">{partner.phone || "-"}</p>
              </div>
              <div>
                <h3 className="text-xs font-semibold text-gray-400 uppercase tracking-wider mb-1">{locale === "th" ? "เข้าร่วมเมื่อ" : "Member Since"}</h3>
                <p className="text-gray-900 font-medium text-sm">{new Date().toLocaleDateString()}</p>
              </div>
            </div>
          </div>
        </div>
      </div>

      {/* Enterprise AI Assessment Card */}
      <div className="bg-white rounded-2xl shadow-sm border border-gray-100 overflow-hidden">
        <div className="px-6 py-5 border-b border-gray-100 bg-gradient-to-r from-slate-50 to-white flex justify-between items-center">
          <h2 className="font-bold text-gray-900 flex items-center gap-2">🤖 CBLUE AI Tier Assessment</h2>
          <span className="text-xs text-gray-500 px-2 py-1 bg-white rounded border border-gray-200">Overall Score: <strong className="text-gray-900">69/100</strong></span>
        </div>
        
        <div className="p-6">
          <div className="flex items-center gap-3 mb-6 p-4 bg-amber-50 rounded-xl border border-amber-100">
            <span className="text-2xl">⚠️</span>
            <div className="flex-1">
              <h4 className="font-bold text-amber-900 text-sm">Partially Verified — Complete profile to improve</h4>
              <p className="text-xs text-amber-700 mt-1">
                Gain more experience, upload portfolio work, update certifications, and maintain good reviews — CBLUE AI will automatically re-evaluate and upgrade your tier.
              </p>
            </div>
          </div>

          <div className="grid grid-cols-1 lg:grid-cols-2 gap-8">
            {/* Evaluation Breakdown */}
            <div>
              <h3 className="text-sm font-bold text-gray-900 mb-4 uppercase tracking-wider">Evaluation Breakdown</h3>
              <div className="space-y-4">
                {[
                  { label: "Experience", score: 25, max: 25, color: "bg-green-500" },
                  { label: "Skills Breadth", score: 12, max: 15, color: "bg-green-500" },
                  { label: "KYC Verification", score: 15, max: 15, color: "bg-green-500" },
                  { label: "Portfolio & Evidence", score: 0, max: 15, color: "bg-gray-200" },
                  { label: "Profile Completeness", score: 7, max: 10, color: "bg-amber-400" },
                  { label: "Price List", score: 6, max: 10, color: "bg-amber-400" },
                  { label: "Credential Verification", score: 4, max: 10, color: "bg-red-400" },
                ].map(item => (
                  <div key={item.label}>
                    <div className="flex justify-between text-xs mb-1">
                      <span className="font-medium text-gray-700">{item.label}</span>
                      <span className="text-gray-500 font-bold">{item.score}/{item.max}</span>
                    </div>
                    <div className="h-1.5 w-full bg-gray-100 rounded-full overflow-hidden">
                      <div className={\`h-full \${item.color}\`} style={{ width: \`\${(item.score / item.max) * 100}%\` }} />
                    </div>
                  </div>
                ))}
              </div>
            </div>

            {/* AI Verification Results */}
            <div>
              <h3 className="text-sm font-bold text-gray-900 mb-4 uppercase tracking-wider">🔍 AI Verification Results</h3>
              <ul className="space-y-3 text-sm">
                <li className="flex items-start gap-2 text-gray-600">
                  <span className="text-red-500 mt-0.5">❌</span>
                  <span>No company info provided</span>
                </li>
                <li className="flex items-start gap-2 text-gray-600">
                  <span className="text-green-500 mt-0.5">✅</span>
                  <span>Experience consistent with project type</span>
                </li>
                <li className="flex items-start gap-2 text-gray-600">
                  <span className="text-red-500 mt-0.5">❌</span>
                  <span>No work description provided</span>
                </li>
                <li className="flex items-start gap-2 text-gray-600">
                  <span className="text-green-500 mt-0.5">✅</span>
                  <span>KYC documents complete (front & back)</span>
                </li>
              </ul>

              <div className="mt-8 p-4 bg-gray-50 rounded-xl text-xs text-gray-500 border border-gray-100">
                <p className="flex items-center gap-2 font-medium text-gray-700 mb-1">
                  <span className="text-lg">🔒</span> Security Notice
                </p>
                Your data is encrypted and protected under PDPA. Credentials are verified to maintain platform integrity.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
`;

code = code.replace(/\/\* ===== PARTNER PROFILE ===== \*\/[\s\S]*?(?=\n}\n\n)/, enhancedProfile + '\n');
fs.writeFileSync('apps/web/app/[locale]/fixers/page.tsx', code);
